core/base/v7/inc/ROOT/TDirectory.h

```diff
@@ -79,10 +77,8 @@ class TDirectory {
   struct ToContentType {
     using decaytype = typename std::decay<T>::type;
     using type = typename std::enable_if<
-      !std::is_function<decaytype>::value
-    && !std::is_pointer<decaytype>::value
-    && !std::is_member_object_pointer<decaytype>::value
-    && !std::is_member_function_pointer<decaytype>::value
+       !std::is_pointer<decaytype>::value
```

Comment:
@karies `std::decay` decays functions to corresponding function pointers. So `is_function::value` is always `false` (and its negation is always `true`) after `decay`.